prow/plugins/projectmanager/projectmanager.go

```diff
@@ -0,0 +1,200 @@
+		Config: map[string]string{"project_manager": `orgsRepos: 
+                             kubernetes/kubernetes:
+                               projects:
+                                 test_project:
+                                   columns:
+                                     - name: triage
+                                     state: open
+                                     org: kubernetes
+                                     labels: 
+                                     - area/conformance
+                                   - id: 0
+                                      name: need-attention
+                                      state: open
+                                      org: kubernetes
+                                      labels: 
+                                      - area/apiserver
+                                 test_project2:
+                                   columns:
+                                   - id: 0
+                                      name: triage
+                                      state: open
+                                      org: kubernetes
+                                      labels:
+                                      - sig/testing
+                                   - id: 0
+                                      name: need-attention
+                                      state: open
+                                      org: kubernetes
+                                      labels: 
+                                      - sig/networking`,
+		},
+	}
+	return pluginHelp, nil
+}
+
+func handlePullRequest(pc plugins.Agent, pe github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.PluginConfig.ProjectManager, pc.Logger, pe)
+}
+
+// Strict subset of *github.Client methods.
+type githubClient interface {
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetRepoProjects(owner, repo string) ([]github.Project, error)
+	GetOrgProjects(org string) ([]github.Project, error)
+	GetProjectColumns(projectID int) ([]github.ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
+}
+
+func handlePR(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, pe github.PullRequestEvent) error {
+	// Only respond to label add or issue/PR open events
+	if pe.Action != github.PullRequestActionOpened &&
+		pe.Action != github.PullRequestActionReopened &&
+		pe.Action != github.PullRequestActionLabeled {
+		return nil
+	}
+	// Get any ManagedProjects that match this PR
+	matchedColumnIDs, err := getMatchingColumnIDs(gc, projectManager, pe)
+	if err != nil {
+		return err
+	}
+	// For each ManagedColumn that matches this PR, add this PR to that Project Column
+	for _, matchedColumnID := range matchedColumnIDs {
+		err = addPRToColumn(gc, matchedColumnID, pe)
+		log.WithError(err).Println("Failed to add PR to project")
```

Comment:
Please use `Error` instead of `Println` and also include relevant log fields using `WithFields` e.g. `pr` and `project-column-id`.

Reply:
Done